services/toDoService.js

```diff
@@ -14,3 +32,59 @@ export async function getAllToDos() {
 
   return sanitizedToDos;
 }
+
+export async function getToDosByDestinationId(destinationId) {
+  if (!validateId(destinationId)) return { status: 400, error: "Invalid destinationId" };
+  await dbConnect();
+
+  const toDos = await ToDo.find({ destinationId: destinationId });
+  if (!Array.isArray(toDos)) return { status: 500, error: "Internal server error" };
+  const sanitizedToDos = toDos.map((toDo) => ({
+    id: toDo.id,
+    description: toDo.description,
+    checked: toDo.checked,
+    destinationId: toDo.destinationId,
+  }));
+
+  return sanitizedToDos;
+}
+
+export async function createToDo(body, destinationId) {
+  if (!validateId(destinationId)) return { status: 400, error: "Invalid destinationId" };
```

Comment:
You should not check for the correct format of the destinationId at all here. The only thing you couls check is if the destination exists in your database.

Reply:
Implemented validity check here as well.